### ISSUE TITLE
Impossible to set the parent window from the render process

### ISSUE THREAD

**Issue by thewizardplusplus** [p-sam/electron-prompt#20]
In this case your check `parentWindow instanceof BrowserWindow` always fail, because `require('electron').remote.getCurrentWindow() !== require('electron').remote.BrowserWindow`. As far as I understand, `getCurrentWindow` returns the wrapper over `BrowserWindow`.

Here's the code to reproduce the bug:

<details>
  <summary>package.json</summary>

```json
{
  "name": "ep-test",
  "version": "1.0.0",
  "main": "main.js",
  "scripts": {
    "start": "electron ."
  },
  "devDependencies": {
    "electron": "^2.0.8"
  },
  "dependencies": {
    "electron-prompt": "^1.1.0-1"
  }
}
```
</details>

<details>
  <summary>main.js</summary>

```js
const {app, BrowserWindow} = require('electron')
const path = require('path')

let window
app.on('ready', () => {
  const preload = path.join(__dirname, 'preload.js')
  const options = {webPreferences: {nodeIntegration: false, preload}}
  window = new BrowserWindow(options)
  window.loadURL('https://electronjs.org/')
  window.on('closed', () => window = null)
  window.webContents.openDevTools()
})
```
</details>

<details>
  <summary>preload.js</summary>

```js
const {getCurrentWindow, BrowserWindow} = require('electron').remote
const prompt = require('electron-prompt')

document.addEventListener('DOMContentLoaded', () => {
  const parent = getCurrentWindow()
  console.log('getCurrentWindow() === BrowserWindow:', parent === BrowserWindow)

  prompt({value: 'test'}, parent)
    .then(result => {
      console.log(...result !== null ? ['result:', result] : ['user canceled'])
    })
    .catch(console.error)
})
```
</details>
<p></p>

Possible solutions:

<details>
  <summary>remove the parent check completely</summary>

```patch
diff --git a/lib/index.js b/lib/index.js
index 207be2c..ba76d74 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -33,11 +33,11 @@ function electronPrompt(options, parentWindow) {
 			width: opts.width,
 			height: opts.height,
 			resizable: opts.resizable,
-			parent: parentWindow instanceof BrowserWindow ? parentWindow : null,
+			parent: parentWindow,
 			skipTaskbar: true,
 			alwaysOnTop: opts.alwaysOnTop,
 			useContentSize: true,
-			modal: Boolean(parentWindow instanceof BrowserWindow),
+			modal: Boolean(parentWindow),
 			title: opts.title,
 			icon: opts.icon
 		});
```
</details>

<details>
  <summary>add the option for disabling the parent check</summary>

```patch
diff --git a/lib/index.js b/lib/index.js
index 207be2c..41fce34 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -33,11 +33,11 @@ function electronPrompt(options, parentWindow) {
 			width: opts.width,
 			height: opts.height,
 			resizable: opts.resizable,
-			parent: parentWindow instanceof BrowserWindow ? parentWindow : null,
+			parent: opts.forceParent || parentWindow instanceof BrowserWindow ? parentWindow : null,
 			skipTaskbar: true,
 			alwaysOnTop: opts.alwaysOnTop,
 			useContentSize: true,
-			modal: Boolean(parentWindow instanceof BrowserWindow),
+			modal: Boolean(opts.forceParent || parentWindow instanceof BrowserWindow),
 			title: opts.title,
 			icon: opts.icon
 		});
```

Then the code `prompt({forceParent: true}, getCurrentWindow())` will be work successfully.
</details>
<p></p>

I'm ready to make a pull request for any solution that you choose.

**Comment by p-sam** [comment]
While it should have been addressed in #4 it appears `require('electron').remote.getCurrentWindow()` is not an instance of `require('electron').remote. BrowserWindow` but rather a plain object with get/set properties. Removing the check seems like the best solution.

**Comment by p-sam** [comment]
fixed by #21, thanks for the PR
